app/models/factory_log.rb

```diff
@@ -2,43 +2,67 @@ class FactoryLog < ActiveRecord::Base
   #TODO: fix algorithm
   #TODO: fix name devide to name and traits
   #TODO: fix visualize association relation and selectable counting or not.
+
+  # Return Factory's log ranking about total time it took.
+  # They are identified factory_name + trait_name.
   def self.time_ranking
     ranking = []
     FactoryLog.all.each do |log|
       if index = ranking.index{|cd| cd[:name] == (log.name + log.traits)}
-        ranking[index][:total_time] += log[:time]
-        ranking[index][:count] += 1
-        ranking[index][:average_time] = ranking[index][:total_time]/ranking[index][:count]
+        hash_data = ranking[index]
+        hash_data[:total_time] += log[:time]
+        hash_data[:count] += 1
+        hash_data[:average_time] = hash_data[:total_time] / hash_data[:count]
       else
-       ranking.push({name: (log.name + log.traits), total_time:log.time, count: 1, average_time: log.time})
+       ranking.push( {
+                      name: (log.name + log.traits),
+                      total_time:log.time,
+                      count: 1,
+                      average_time: log.time
+                      } )
      end
     end
     ranking.sort_by!{|l| l[:total_time]}.reverse
   end
 
-  def self.average_time(factory)
-    sum_time(factory) / log_set_with_trait(factory).size
+  # Return specified factory's average exectuaion time.
+  def self.average_time(name_and_traits)
+    logs = FactoryLog.log_set_with_trait(name_and_traits)
+    logs.map{ |log| log.time }.inject{ |sum, n| sum + n } / logs.size
   end
 
-  def self.sum_time(factory)
-    log_set_with_trait(factory).map{|f| f.time}.inject{|fl1, fl2| fl1 + fl2 }
+  # Return specified factory's total exectuaion time.
+  def self.sum_time(name_and_traits)
+    log_set_with_trait(name_and_traits).map(&:time).inject do |log1, log2|
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'log1'
* [UncommunicativeVariableName] has the variable name 'log2'